src/pixelmapFeature.js

```diff
@@ -134,6 +116,42 @@ var pixelmapFeature = function (arg) {
     return m_this;
   };
 
+  /**
+   * Mark that an index's data value (and hence its color) has changed without
+   * marking all of the data array as changed.  If this function is called
+   * without any parameters, it clears the tracked changes.
+   *
+   * @param {number} [idx] The lowest data index that has changed.  If
+   *    `undefined`, return the current tracked changed range.
+   * @param {number|'clear'} [idx2] If an index was specified in `idx` and
+   *    this is specified, the highest index (inclusive) that has changed.  If
+   *    returning the tracked changed range and this is `clear`, clear the
+   *    tracked range.
+   * @returns {this|number[]} When returning a range, this is the lowest and
+   *    highest index values that have changed (inclusive), so their range is
+   *    `[0, data.length)`.
+   */
+  this.indexModified = function (idx, idx2) {
+    if (idx === undefined) {
+      const range = m_modifiedIndexRange;
+      if (idx2 === 'clear') {
+        m_modifiedIndexRange = undefined;
+      }
+      return range;
+    }
+    m_this.modified();
+    if (m_modifiedIndexRange === undefined) {
+      m_modifiedIndexRange = [idx, idx];
+    }
+    if (idx < m_modifiedIndexRange[0]) {
+      m_modifiedIndexRange[0] = idx;
+    }
+    if ((idx2 || idx) > m_modifiedIndexRange[1]) {
+      m_modifiedIndexRange[1] = (idx2 || idx);
+    }
```

Comment:
What are the downstream consequences of calling code not abiding by the `idx < idx2` constraint? Is there a need to protect against this scenario when building `m_modifedIndexRange`?

Reply:
If you pass idx2 < idx, the resulting tracked range will be `min(m_modifiedIndexRange[1], idx), max(m_modifiedIndexRange[2], idx2)`.  In the case where the index range had not previously been set, the functional result will be that no update will be required.  If the index range has previously been set, then this will possibly extend the lower end of the range but not extend the upper end.

I don't think we need to guard against it, as it isn't clear what the caller would expect if asking to update from an inverted range.  Nothing untoward will happen -- it just won't update what they expect.

There are many functions that don't strictly check input -- for instance, the indices are expected to be integers, but we don't confirm that, and, if idx is not an integer, the program throw an error.  My inclination is to leave it without an explicit check.

Reply:
That seems reasonable to me.